src/figuresClasses.ts

```diff
@@ -1,19 +1,83 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
 
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle'
 }
 
-export class Triangle {
-
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape = Shape.triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Length of sides must be greater than 0.');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('Invalid triangle sides.'
+      + 'Sides 1, 2 and 3 can\'t form a triangle');
+    }
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
```

Comment:
```suggestion
  getArea(): number {
    const {a, b, c} = this;
    
    const p = (a + b + c) / 2;
    const area = Math.sqrt(p * (p - a) * (p - b) * (p - c));
```
destructuring this little decrease amount of code